fml/memory/task_runner_checker.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// A class for checking that the current thread is/isn't the same as an initial
+// thread.
+
+#ifndef FLUTTER_FML_MEMORY_TASK_RUNNER_CHECKER_H_
+#define FLUTTER_FML_MEMORY_TASK_RUNNER_CHECKER_H_
+
+#include "flutter/fml/message_loop.h"
+#include "flutter/fml/task_runner.h"
+
+namespace fml {
+
+class TaskRunnerChecker final {
+ public:
+  TaskRunnerChecker() : self_(InitTaskQueueId()){};
+
+  ~TaskRunnerChecker() = default;
+
+  bool RunsOnCreationTaskRunner() const {
+    FML_DCHECK(fml::MessageLoop::IsInitializedForCurrentThread());
+
+    const auto current_queue_id = MessageLoop::GetCurrentTaskQueueId();
+
+    if (current_queue_id == self_) {
+      return true;
+    }
+
+    auto queues = MessageLoopTaskQueues::GetInstance();
+    if (queues->Owns(current_queue_id, self_)) {
+      return true;
+    }
+    if (queues->Owns(self_, current_queue_id)) {
+      return true;
+    }
+    return false;
+  }
+
+ private:
+  TaskQueueId self_;
```

Comment:
I think `self_` is an ambiguous name. Can this be called, `initialized_queue_id_`?